x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -61,6 +62,121 @@ func RandomMsgCreateConcentratedPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx
 	}, nil
 }
 
+func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCreatePosition, error) {
+	// get random pool
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// get random user address with the pool denoms
+	sender, tokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we always have 2 tokens
+	// Note: tokens returns a random subset of poolDenoms, so  had to add this assertion
+	if len(tokens) < 2 {
+		return nil, fmt.Errorf("user doesnot have pool tokens")
+	}
+
+	//  Retrieve minTick and maxTick from precision factor
+	minTick, maxTick := clkeeper.GetMinAndMaxTicksFromExponentAtPriceOne(clPool.GetPrecisionFactorAtPriceOne())
+
+	// Randomize lowerTick and upperTick from max values to create position
+	lowerTick, upperTick, err := getRandomTickPositions(sim, minTick, maxTick, clPool.GetTickSpacing())
+	if err != nil {
+		return nil, err
+	}
+
+	return &cltypes.MsgCreatePosition{
+		PoolId:        clPool.GetId(),
+		Sender:        sender.Address.String(),
+		LowerTick:     lowerTick,
+		UpperTick:     upperTick,
+		TokenDesired0: tokens[0],
+		TokenDesired1: tokens[1],
+		// TODO: Randomize TokenMinAmount0 and TokenMinAmount1 in next iteration
+		TokenMinAmount0: sdk.NewInt(0),
+		TokenMinAmount1: sdk.NewInt(0),
+	}, nil
+}
+
+func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
+	rand := sim.GetSeededRand("select random position")
```

Comment:
Suggest doing this in all cases because I don't think we should reuse the same domain separation across the messages
```suggestion
	rand := sim.GetRand()
```

Reply:
Discussed this on Slack, it looks like we shouldn't share seeds between messages to avoid an error in one function drastically changing randomness in another. So I think `GetRand` is preferable here because it doesn't use shared domain

Reply:
makes sense! updated